CMake not working for windows

The CMake gives the error 
```bash
-- The CXX compiler identification is unknown
CMake Error at CMakeLists.txt:2 (project):
  The CMAKE_CXX_COMPILER:

    cl

  is not a full path and was not found in the PATH.

  To use the NMake generator with Visual C++, cmake must be run from a shell
  that can use the compiler cl from the command line.  This environment is
  unable to invoke the cl compiler.  To fix this problem, run cmake from the
  Visual Studio Command Prompt (vcvarsall.bat).

  Tell CMake where to find the compiler by setting either the environment
  variable "CXX" or the CMake cache entry CMAKE_CXX_COMPILER to the full path
  to the compiler, or to the compiler name if it is in the PATH.


-- Configuring incomplete, errors occurred!
See also "C:/Users/Harshit_Singh/Documents/my doc/linear_algebra/Matrix/build/CMakeFiles/CMakeOutput.log".
See also "C:/Users/Harshit_Singh/Documents/my doc/linear_algebra/Matrix/build/CMakeFiles/CMakeError.log".
```
but the file compiles when the file is compiled using the following code:
```bash
 g++ examples/abs.cpp -std=c++11 -I./include/ -o ./build/abs
```

This issue occurs because Windows defaults to NMake Generator (which is part of Windows build tools). 

A quick workaround would be delete the `build/` folder and then rerun the commands in README with cmake as:
`cmake -G "MinGW Makefiles" ../`

I will send in a PR to make the necessary changes so that it default to `mingw-make`. 

@harshitsingh85420 Do you have `make` installed? 
You can test it by running `mingw32-make` or `mingw-make` on 32-bit and 64-bit windows respectively.  

A sample output when `mingw-make` is installed and run in the `Matrix/` folder would be: 
```
$ mingw-make
mingw-make: *** No targets specified and no makefile found.  Stop.
```

--- a/PreLoad.cmake
+++ b/PreLoad.cmake
@@ -1,0 +1,3 @@
+if(WIN32)
+    set (CMAKE_GENERATOR "MinGW Makefiles" CACHE INTERNAL "" FORCE)
+endif(WIN32)

--- a/README.md
+++ b/README.md
@@ -32,6 +32,16 @@ On a unix system, the build directory should now look something like this:
 
 ## Development
 
+### Linux
+
+#### Prerequisites
+
+1. C++ compiler (gcc or clang)
+2. git
+3. clang-format
+5. make (for Makefiles)
+4. cmake (version 3.13 or higher)
+
 ```bash
 # Check out the library.
 $ git clone https://github.com/mlcpp/Matrix.git && cd Matrix
@@ -50,6 +60,40 @@ $ make
 
 # Compile single file and place the binary inside build/examples/ folder.
 $ make <file_name_without_extension>
+```
+
+### Windows
+
+#### Prerequisites
+
+1. C++ compiler (mingw-gcc)
+2. git (msysgit)
+3. clang-format
+5. mingw-make or mingw32-make (for Makefiles)
+4. cmake (version 3.13 or higher)
+
+Add these tools to PATH. 
+
+__Note__: Run the following commands in msysgit terminal. 
+
+```bash
+# Check out the library.
+$ git clone https://github.com/mlcpp/Matrix.git && cd Matrix
+
+# Create build directory.
+$ mkdir build
+
+# Enter into build directory.
+$ cd build
+
+# Run CMake. This step will generate a Makefile in the build/ folder.
+$ cmake ../
+
+# Compile all files and place the binaries inside build/examples/ folder.
+$ mingw-make                                          # use mingw32-make if using 32-bit Windows
+
+# Compile single file and place the binary inside build/examples/ folder.
+$ mingw-make <file_name_without_extension>            # use mingw32-make if using 32-bit Windows
 ```
 
 

